### PR TITLE
Updates API

### PR DESCRIPTION
**Pull request by TroyDota**
Updates the golang stuct to current api documentation and adds new util

### CODE REVIEW

File: util.go [hunk 1]
```diff
@@ -15,3 +15,13 @@ func SnowflakeTimestamp(ID string) (t time.Time, err error) {
 	t = time.Unix(0, timestamp*1000000)
 	return
 }
+
+//PermissionsSynced Checks if channel has permissions synced with another channel
+func PermissionsSynced(c1 *Channel, c2 *Channel) bool {
```
Comment (TroyDota) [inline util.go:20]: I am not sure if this is wanted or not?

Comment (CarsonHoffman) [reply util.go:20]: I think it'd be best to leave this out; two channels being synced doesn't have a strong definition in the API; the `PermissionOverwrites` could have different orders, and without robust tests for the function I think that it'd be best to defer to the "lower-level" promise of DiscordGo and not include it.

**Comment by CarsonHoffman** [comment]
Hi! Thank you for submitting this PR; in accordance with the [Contributing](https://github.com/bwmarrin/discordgo#contributing) section of the README, we'd prefer that any PRs are opened against the `develop` branch rather than `master`. You should be able to change the target branch in the PR, however please also rebase your `minor-updates` branch onto our `develop` branch before doing so, or else any differences between `master` and `develop` will also be brought into this PR.

**Comment by TroyDota** [comment]
Oh sorry my bad.

**Comment by TroyDota** [comment]
Did i mess it up?

**Comment by CarsonHoffman** [comment]
It looks like you didn't rebase your branch onto `develop`; notice that a commit that I had submitted to `master` is now part of the PR.

**Comment by TroyDota** [comment]
Yeah i saw sorry, ill rebase.

**Comment by TroyDota** [comment]
I think we coo now?

**Comment by TroyDota** [comment]
Sorry im not clued up with git commands idk if there is a better way.

File: util.go (hunk 1 above)
Comment (TroyDota) [reply util.go:20]: Okay

Comment (TroyDota) [reply util.go]: One last thing, i added this for the https://i.nuuls.com/tVWo9.png

**Comment by TroyDota** [comment]
I am not sure if this is all the minor api changes, its the ones that i have used in my bot.

File: structs.go [hunk 2]
```diff
@@ -492,8 +530,46 @@ type Guild struct {
...
+)
+
+// SystemChannelFlag is the type of flags in the system channel (see SystemChannelFlag* consts)
+// https://discord.com/developers/docs/resources/guild#guild-object-system-channel-flags
+type SystemChannelFlag int
+
+// Block containing known SystemChannelFlag values
+const (
+	SystemChannelFlagsSuppressJoin    SystemChannelFlag = 1 << 0
+	SystemChannelFlagsSuppressPremium SystemChannelFlag = 1 << 1
```
Comment (CarsonHoffman) [inline structs.go:570]: Same as above

File: structs.go [hunk 3]
```diff
@@ -445,6 +468,12 @@ type Guild struct {
 	// update events, and thus is only present in state-cached guilds.
 	Presences []*Presence `json:"presences"`
 
+	// the maximum number of presences for the guild (the default value, currently 25000, is in effect when null is returned)
+	MaxPresences int `json:"max_presences"`
+
+	// the maximum number of members for the guild
```
Comment (CarsonHoffman) [inline structs.go:474]: Comments should start with Capital

File: structs.go [hunk 4]
```diff
@@ -492,8 +530,46 @@ type Guild struct {
...
+	// the maximum amount of users in a video channel
+	MaxVideoChannelUsers int `json:"max_video_channel_users"`
+
+	// approximate number of members in this guild, returned from the GET /guild/<id> endpoint when with_counts is true
+	ApproximateMemberCount int `json:"approximate_member_count"`
+
+	// approximate number of non-offline members in this guild, returned from the GET /guild/<id> endpoint when with_counts is true
+	ApproximatePresenceCount int `json:"approximate_presence_count"`
+
+	// permissions of our user
```
Comment (CarsonHoffman) [inline structs.go:549]: Comments should start with Capital

File: structs.go (hunk 3 above)
Comment (TroyDota) [reply structs.go:474]: 👍

**Comment by TroyDota** [comment]
👍 i looked at integrations too. Thought might aswell go through the other stuff and make sure the discordgo was consistent with the current api.

**Comment by TroyDota** [comment]
yeah have fun going through it <3

**Comment by TroyDota** [comment]
Does it fix #386 ?

File: message.go [hunk 5]
```diff
@@ -117,9 +120,22 @@ type Message struct {
...
+// https://discord.com/developers/docs/resources/channel#message-object-message-flags
+type MessageFlags int
+
+// Valid MessageFlags values
+const (
+	MessageFlagsCrossPosted MessageFlags = 1 << iota
+	MessageFlagsIsCrossPosted
+	MessageFlagsSupressEmbeds
+	MessageFlagsSourceMessageDeleted
+	MessageFlagsUrgant
```
Comment (CarsonHoffman) [inline message.go:136]: "Urgant" -> "Urgent"

**Comment by CarsonHoffman** [comment]
I went ahead and rebased the changes onto the `develop` branch for you; your attempt at reverting a commit as a rebase unfortunately left your branch in a state that was unable to be merged. Anyway, thanks for all the extra functionality that this PR brings to the library!

**Pull request merged**